app/main.py

```diff
@@ -2,24 +2,69 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.max_amount = max_amount
+        self.min_amount = min_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: object) -> str:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError
```

Comment:
The `TypeError` raised here should include an informative message, e.g., `raise TypeError('Value must be an integer')`.